flang/include/flang/Lower/OpenMP.h

```diff
@@ -27,8 +29,9 @@ namespace pft {
 struct Evaluation;
 } // namespace pft
 
+using GenFIRCBTy = std::function<void(Fortran::lower::pft::Evaluation &eval)>;
 void genOpenMPConstruct(AbstractConverter &, pft::Evaluation &,
-                        const parser::OpenMPConstruct &);
+                        const parser::OpenMPConstruct &, GenFIRCBTy genFIRCB);
```

Comment:
Using Callbacks approach has following advantages:
1. No duplication, otherwise might need to pull out lot of code from `Bridge.cpp`(where most of the lowering code is residing).
2. Less error prone, since for actual codegen we fall back to `Flang` codegen cleanly.
3. Code flow and interfacing of `OpenMP` codegen from rest of the codegen remains pretty independent/good/.